program/llvm-toolchain-ppl.md

```diff
@@ -0,0 +1,252 @@
+
+## Industry landscape
+
+The "original" compiler used for the development of RISC-V cores is GCC.  However, LLVM offers a licensing scheme that is more appealing to the processor industry as it does not require the disclosure of critical vendor-specific intellectual property (IP) when distributing the software.  LLVM-based compiler chains appear to be at the foundation of many market leader offerings and it is expected that this will also be the case for RISC-V / CORE-V solutions.
+
+### Related efforts to be described
+
+The upstream LLVM tool chain project already supports standard RISC-V, including all ratified ISA extensions.  RISC-V support in LLVM is strongly compliant to the RISC-V specification, but relatively immature in comparison to other architectures as x86, Arm and MIPS.
+
+Currently LLVM does not support CORE-V specific features or extensions, and subsequently, its test suites do not include any CORE-V or PULP-specific tests.
+
+Other toolchains available for RISC-V are:
+- the GNU tool chain for RISC-V, for which a CORE-V specific project is being set up within OpenHW Group;
+- IAR RISC-V compiler
+
+The GNU tool chain project within OpenHW Group intends to support CORE-V and PULP extensions, but at the time of writing focuses solely on the CV32E40P platform.
+
+An OHG architecture project devoted to the CVA6 (Ariane) family of cores has passed the PPL gate.  There is a clear synergy between this project and the CVA6 project:
+- CVA6 hardware design and verification will directly benefit from the LLVM work;
+- in return, the CVA6 project will directly provide inputs (specifications, documentation) for the compiler- and simulator-related activities in the present project.
+
+## External dependencies
+
+Prequisites:
+- all the upstream tool sources
+- a suitable platform for regression testing the compiler.
+- agreement on the instruction set encodings to be compliant with the RISC-V standard.
+
+External dependencies:
+- ongoing tracking of upstream LLVM development until the CORE-V tool chain is accepted upstream.
+
+## List of project outputs (deliverables)
+
+## Project deliverables
+
+### Final deliverables
+
+1. extensions to upstream LLVM compiler tools to support CORE-V;
+2. revisions to the CORE-V design specifications to clarify ambiguities.
+
+### Interim deliverables
+
+1. Reports on progress to the monthly SW TG:
+   - progress against work packages;
+   - regression test results;
+   - updates to the project plan; and
+   - updates to the risk register.
+2. Continuously updated source code as new features are added.
+
+## TGs impacted/resource requirements
+
+The Software TG will be responsible for oversight of the planning and delivery of this project.
+
+## OpenHW engineering staff resource plan: requirement and availability
+
+- Duncan Bees - program management oversight
+
+## Engineering resource supplied by members - requirement and availability
+
+TBD
```

Comment:
Embecosm: two engineer months from Philipp Krones in Q4/2020 for CV32E40P support.